cmd/swarm/main.go

```diff
@@ -252,6 +252,10 @@ func registerBzzService(ctx *cli.Context, stack *node.Node) {
 	if len(bzzport) > 0 {
 		bzzconfig.Port = bzzport
 	}
+	networkId := ctx.GlobalUint64(SwarmNetworkIdFlag.Name)
+	if networkId > 0 {
```

Comment:
the problem is we assign the default value to this so its never gonna be 0
https://github.com/jmozah/go-ethereum/blob/b30aab463831634a368e925a0fb87d6a828bf02d/cmd/swarm/main.go#L79
so best to delete that line
and check in api/config that if network id is 0 then take default, `network.NetworkId` constant
same with other flags  like port possibly